stores/sqlstore_test.go

```diff
@@ -0,0 +1,1209 @@
+// Copyright 2017 Apcera Inc. All rights reserved.
+
+package stores
+
+import (
+	"database/sql"
+	"fmt"
+	"math/rand"
+	"reflect"
+	"regexp"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/nats-io/nats-streaming-server/spb"
+	"github.com/nats-io/nats-streaming-server/test"
+)
+
+// The SourceAdmin is used by the test setup to have access
+// to the database server and create the test streaming database.
+// The Source contains the URL that the Store needs to actually
+// connect to the server and use the database.
+const (
+	testDefaultDatabaseName = "test_nats_streaming"
+
+	testDefaultMySQLSource      = "nss:password@/" + testDefaultDatabaseName
+	testDefaultMySQLSourceAdmin = "nss:password@/"
+
+	testDefaultPostgresSource      = "dbname=" + testDefaultDatabaseName + " sslmode=disable"
+	testDefaultPostgresSourceAdmin = "sslmode=disable"
+)
+
+var (
+	testSQLDriver       = driverMySQL
+	testSQLDatabaseName = testDefaultDatabaseName
```

Comment:
I would consider using a timestamp appended to the name by default. This can aid in debugging. Keeps things clean on good runs. Ensures db doesn't get overwritten on bad runs.

Reply:
Not sure I follow here. The same database is reused. Believe me, it speeds up tremendously the runs especially with Postgres. The tables are cleaned-up between each run.
If there was a test that would start to fail (after a code change), I could disable the cleanup of the database should I inspect it. Write the tests in the optic that they are passing, not that they may fail.